### PR TITLE
Feature/s3 secure

### PR DESCRIPTION
**Pull request by sfriedlizuehlke**
Add the config option `secure` under the key `staging.s3`. If set to true it will use an https connection to the S3 stoage.

### CODE REVIEW

File: polytope_server/common/staging/s3_staging.py [hunk 1]
```diff
@@ -47,15 +47,16 @@ def __init__(self, config):
         access_key = config.get("access_key", "")
         secret_key = config.get("secret_key", "")
         self.bucket = config.get("bucket", "default")
+        use_secure = config.get("use_secure", False) == True
```
Comment (cfkanesan) [inline polytope_server/common/staging/s3_staging.py]: The name could be just secure to remain consistent with the other parameters

File: tests/unit/test_s3_staging.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+@mock.patch('polytope_server.common.staging.s3_staging.Minio', autospec=True)
+def test_s3_staging_secure_true(mock_minio: mock.Mock):
+   s3Staging = S3Staging(config={'use_secure': True})
+
+   verify_secure_flag_and_internal_url(mock_minio, s3Staging, True)
+
+def verify_secure_flag_and_internal_url(mock_minio: mock.Mock, s3Staging: S3Staging, secure: bool):
+      mock_minio.assert_called_once()
+      _, kargs = mock_minio.call_args
```
Comment (cfkanesan) [inline tests/unit/test_s3_staging.py]: nit: the conventional name is `kwargs`

**Review by cfkanesan — APPROVED** [review]
Changes look good. Could you maybe enable black formatting or at least run it on the new test module?

File: tests/unit/test_s3_staging.py (hunk 2 above)
Comment (sfriedlizuehlke) [reply tests/unit/test_s3_staging.py]: 6078d9c

File: polytope_server/common/staging/s3_staging.py (hunk 1 above)
Comment (sfriedlizuehlke) [reply polytope_server/common/staging/s3_staging.py]: 9566e33

**Review by victoria-cherkas — APPROVED** [review]

**Review by cfkanesan — APPROVED** [review]

**Pull request merged**